### ISSUE TITLE
MIPS Relocations Regression 

### ISSUE THREAD

**Issue by bpotchik** [Vector35/binaryninja-api#6183]
**Version and Platform (required):**
- Binary Ninja Version: 4.2.6455

**Bug Description:**
Broken MIPs relocations:

From previous stable (4.1.5902)
`0006a648  08965226   addiu   $s2, $s2, -0x69f8`

New Stable (4.2.6455)
`0006a648  00005226   addiu   $s2, $s2, 0`

Search for "build adapt overcome thrive" for local file repro.

**Comment by bpotchik** [comment]
Also, reverting the recent relocation related changes in arch_mips.cpp resolves the issue:
`git checkout 590b594850ccd15b9e2cb2b0e8f3050d782647b3 -- arch_mips.cpp`

**Comment by zznop** [comment]
I reverted the code handling R_MIPS_HI16 / R_MIPS_LO16 relocations that was changed as part of this PR:

https://github.com/Vector35/binaryninja-api/pull/5919

It appears to have fixed the regression (in build `4.3.6488-dev`).

**Comment by fuzyll** [comment]
These are the commits we'll need if we're cutting a "service pack" release:

```
commit e0c57c856b6ae83851bd75af1fb327bbb05a5c93
commit 83ba34da6a0c0e4a5bdf7fca111d13bc8938c9c3
```
